src/server/scripts/EasternKingdoms/ZulGurub/boss_thekal.cpp

```diff
@@ -153,15 +153,23 @@ class boss_thekal : public CreatureScript
 
         void DamageTaken(Unit* /*attacker*/, uint32& damage, DamageEffectType, SpellSchoolMask) override
         {
+            LOG_DEBUG("scripts.ai", "boss_thekalAI::DamageTaken(): {} received {} damage!",
+                me->GetName(), damage
+            );
+
             if (!me->HasAura(SPELL_TIGER_FORM) && damage >= me->GetHealth())
             {
+                LOG_DEBUG("scripts.ai", "boss_thekalAI::DamageTaken(): {} received a killing blow!",
```

Comment:
same here

Reply:
I can remove these, but I'd be interested to know if there's guidance somewhere on when logs are or aren't appropriate. I can see an argument for the on-every-hit log, but the "received a killing blow" log is pretty critical to how this AI works.

Can you help me understand under what criteria you consider a logging statement correct?

Reply:
Personally, I would say logs are appropriate in functions/methods where they are broader, not in individual scripts. Obviously, there are some exceptions.

Also, I don't see this log ever producing anything useful.

Reply:
I guess it's a philosophical debate. I think of debug logs as useful for when something is broken and I need to fix it - in that case, this log would tell you the boss just triggered this part of the AI code. If you're not actively debugging `scripting.ai`, you'd never know it was there. 

Regardless, you're the one approving, so I can change it if that's what needed for this PR to be useful.